### PR TITLE
feat: shooting computer fields

### PR DESCRIPTION
**Pull request by JustD1d1t**
Add:
- logic for shooting computer fields
- removing objects from arrays with ships/empty fields
- changing computer state after shoot
- overlay for player board

### CODE REVIEW

File: src/game/playerShooting.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+// import { computerState, computerShips, computerEmptyFields } from "./game";
```
Comment (ruljin) [inline src/game/playerShooting.js]: If something is commented that it is unneeded - remove it

File: src/game/playerShooting.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  );
+  if (fieldindex >= 0) {
+    const shipName = computerShips[fieldindex].shipName;
+    computerShips.splice(fieldindex, 1);
+    let ship = computerState.ships.find((ship) => ship.shipName === shipName);
+    let field = ship.fields.findIndex((field) => field.name === fieldName);
+    ship.fields[field].hit = true;
+    const isSunk = ship.fields.every((field) => field.hit === true);
+    ship.sunk = isSunk;
+    const lostAllShieps = computerState.ships.every(
```
Comment (ruljin) [inline src/game/playerShooting.js]: `lostAllShips`

File: src/game/playerShooting.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      (ship) => ship.sunk === true
```
Comment (ruljin) [inline src/game/playerShooting.js]: Just `ship.sunk` 😄 ?

File: src/game/playerShooting.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    );
+    computerState.lostAllShips = lostAllShieps;
+  } else {
+    fieldindex = computerEmptyFields.findIndex(
+      (field) => field.fieldName === fieldName
+    );
+    if (fieldindex >= 0) {
+      computerEmptyFields.splice(fieldindex, 1);
+    } else return;
```
Comment (ruljin) [inline src/game/playerShooting.js]: You can leave `return` in that else or just move it at the end of the function and pass `return` as the end on function ... even though you are not returning anything so is it needed even?

File: src/game/playerShooting.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  }
+}
+
+export function addShooting() {
```
Comment (ruljin) [inline src/game/playerShooting.js]: `addShooting` sounds a bit weird - what does this method do? It adds a field possibility to be targeted or what?

**Review by ruljin — CHANGES_REQUESTED** [review]

File: src/game/playerShooting.js (hunk 1 above)
Comment (JustD1d1t) [reply src/game/playerShooting.js]: :+1:

File: src/game/playerShooting.js (hunk 2 above)
Comment (JustD1d1t) [reply src/game/playerShooting.js]: here we go again :D funciton

File: src/game/playerShooting.js (hunk 3 above)
Comment (JustD1d1t) [reply src/game/playerShooting.js]: I don't understand. I check is every ship sunk then return true or false

File: src/game/playerShooting.js (hunk 4 above)
Comment (JustD1d1t) [reply src/game/playerShooting.js]: Yeah, it's unnecessary. I could just use if statement without else

File: src/game/playerShooting.js (hunk 5 above)
Comment (JustD1d1t) [reply src/game/playerShooting.js]: It means that i add possibility for field to be targeted. Maybe `addPossibilityToTargetField` ? In my opinion it's a bit long. Have you got any ideas to call this function?

Comment (ruljin) [reply src/game/playerShooting.js]: `makeFieldTargetable`

**Review by ruljin — APPROVED** [review]

**Pull request merged**